babybuddy/middleware.py

```diff
@@ -88,3 +90,38 @@ class CustomRemoteUser(RemoteUserMiddleware):
     """
 
     header = getenv("PROXY_HEADER", "HTTP_REMOTE_USER")
+
+
+class HomeAssistant:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.use_x_ingress_path_rewrite = settings.HOME_ASSISTANT_USE_X_INGRESS_PATH
+
+    def __call__(self, request: HttpRequest):
+        def wrap_x_ingress_path(org_func):
+            if not request.is_homeassistant_ingress_request:
+                return org_func
+            x_ingress_path = request.headers.get("X-Ingress-Path")
+            if x_ingress_path is None:
+                return org_func
+
+            def wrapper(*args, **kwargs):
+                url = org_func(*args, **kwargs)
+
+                url_parts = urlsplit(url)
+                url = urlunsplit(
+                    url_parts._replace(path=x_ingress_path + url_parts.path)
+                )
+
+                return url
+
+            return wrapper
+
+        request.is_homeassistant_ingress_request = (
+            request.headers.get("X-Hass-Source") == "core.ingress"
+        )
+
+        if self.use_x_ingress_path_rewrite:
```

Comment:
Is there any reason one would _not_ want this functionality if we know this is an ingress request? I.e., could we not just rely on the `request.is_homeassistant_ingress_request` instead of a requiring a bespoke configuration toggle?

Reply:
Ok, this is all makes sense.

> ... now that I wrote this down however, I realize that this is not quite what I implemented. The check if a request is an ingress request is happening regardless of the "active state" of the middle ware. I will fix that one. That one should also be

Yeah I think this is part of where I was confused 😄

Reply:
Fixed!